CMakeLists.txt

```diff
@@ -18,6 +18,20 @@ set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 # use our own version of FindBoost.cmake and other Find* scripts
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
+# function to remove a flag from a variable
+function(RemoveFlag VAR SCOPE FLAG)
+	MESSAGE("Removing ${FLAG} flag from ${VAR}")
+	separate_arguments(${VAR})
+	list(REMOVE_ITEM ${VAR} ${FLAG})
+	string(REPLACE ";" " " ${VAR} "${${VAR}}")
+	
+	if("${SCOPE}" STREQUAL "CACHE")
+		set(${VAR} "${${VAR}}" CACHE STRING "" FORCE)
+	elseif("${SCOPE}" STREQUAL "PARENT")
```

Comment:
This scope is the local scope from the POV of the caller. I'd prefer if you called this scope "LOCAL" or something.

Reply:
What about this?
```
if("${SET_IN}" STREQUAL "CACHE")
	set(${VAR} "${${VAR}}" CACHE STRING "Removed ${FLAG}" FORCE)
elseif("${SET_IN}" STREQUAL "SCRIPT")
	set(${VAR} "${${VAR}}" PARENT_SCOPE)
endif()
```

Reply:
"SCRIPT" would be fine for me.